connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -154,7 +154,7 @@ public void onShutdown(String connector) {
 
     @Override
     public void onFailure(String connector, Throwable cause) {
-        statusBackingStore.putSafe(new ConnectorStatus(connector, ConnectorStatus.State.FAILED,
+        statusBackingStore.put(new ConnectorStatus(connector, ConnectorStatus.State.FAILED,
```

Comment:
Previously `putSafe` was okay because connectors & tasks would always have some other state like `STARTED`, but it's possible now to transition directly to `FAILED`.